website/docs/advanced-concepts/access-control/roles.md

```diff
@@ -0,0 +1,237 @@
+
+
+# Roles
+ A Role is a collection of permissions that defines access for entities or resources which can be further assigned to a user or a group. Roles allow you to adjust multiple permissions rather than assigning individual permissions to users/groups. In Appsmith, a role can contain multiple permissions and a permission can be present in multiple roles. However, a role can't be nested, i.e, it can contain only permissions, not other roles. 
```

Comment:
It sounds a bit confusing to me, as roles will also define actions that users can take and not just access right. Can we rewrite it like as below:
```suggestion
 In Granular Access Control (GAC), a role is a defined set of permissions that can be assigned to a user or group of users. These permissions determine the actions that can be performed on specific entities or resources. By using roles, you can manage multiple permissions at once rather than managing them individually. In Appsmith, a role can be mapped to multiple permissions, and single permission can be mapped to multiple roles. However, a role cannot be nested, meaning that it exists as a standalone entity with its own defined set of permissions. Roles are useful for efficiently managing access and permissions within your Appsmith account.
```

Reply:
Thanks @jnikhila . Permission defines access to the actions that a user can take. They do not define the actions directly but give access to a user to perform certain actions. Also, just for consistency with the UI, let's keep it as "assigned to a user or a group."

Reply:
Ok here let's avoid 'actions' and use 'operations' instead . Yes agree it need to be clear . suggestion below including some points shared by Nikhila above

```suggestion
A Role is a collection of permissions that grant access to perform certain operations on resources and can be assigned to a user or a group. Roles enable users to manage multiple permissions rather than assigning individual permissions to users/groups. In Appsmith, a role can be mapped to multiple permissions and a permission can be mapped to multiple roles. However, a role cannot be nested, i.e. it can contain only permissions and not other roles. Roles are useful to provide fine-grained access and for managing permissions on Appsmith resources.
```